src/main/kotlin/com/ampnet/walletservice/grpc/blockchain/BlockchainServiceImpl.kt

```diff
@@ -68,7 +69,12 @@ class BlockchainServiceImpl(
             response.balance.toLongOrNull()
         } catch (ex: StatusRuntimeException) {
             logger.warn("Could not get balance for wallet: $hash", ex)
-            null
+            val grpcException =
+                generateInternalExceptionFromStatusException(ex, "Could not get balance for hash: $hash")
+            when (grpcException) {
+                is GrpcHandledException -> return null
+                else -> throw grpcException
+            }
```

Comment:
For known errors service will return balance `null`, for unknown response will be `502`